### ISSUE TITLE
Possible bugs in the DQD field level checks

### ISSUE THREAD

**Issue by francescopozzoni** [OHDSI/DataQualityDashboard#368]
Hi everyone,
I found these two behaviour of the DQD and I think they can be potential bugs. I am running my dqd on a cdm v5.4 on a postgresql server.

This check seem to treat the specimen quantity field as a number of sample instead of the actual quantity of specimen stored. Is it correct? The low level range value should be 0 because a value of 0.1 ml for example should be accepted.

![fail-specimen](https://user-images.githubusercontent.com/69623478/193615446-4e214d15-b570-4c36-885c-74b3e3a07c6b.PNG)

```
/*********
PLAUSIBLE_VALUE_LOW
get number of records and the proportion to total number of eligible records that fall below this threshold
Parameters used in this template:
cdmDatabaseSchema = cdm1
cdmTableName = SPECIMEN
cdmFieldName = QUANTITY
plausibleValueLow = 1
**********/
SELECT num_violated_rows, CASE WHEN denominator.num_rows = 0 THEN 0 ELSE 1.0*num_violated_rows/denominator.num_rows END  AS pct_violated_rows, 
  denominator.num_rows as num_denominator_rows
FROM
(
	SELECT COUNT(violated_rows.violating_field) AS num_violated_rows
	FROM
	(
		SELECT 'SPECIMEN.QUANTITY' AS violating_field, cdmTable.*
		from cdm1.SPECIMEN cdmTable
      where cdmTable.QUANTITY < 1
	) violated_rows
) violated_row_count,
(
	SELECT COUNT(*) AS num_rows
	FROM cdm1.SPECIMEN cdmTable
	where QUANTITY is not null
) denominator
;
```

The second check is this one on the Domain of the event_object_concept_id. This check is bugged for the sql query that is run (the IN statement should hava a list of the admitted domain values and not a single string), but the other issue is that this query also triggers a fail episodes that are Diesease Episode that as the documentation says should have an event_object_concept_id that comes from the Condition domain.

![fail-episode](https://user-images.githubusercontent.com/69623478/193615370-ca7304ba-685d-47d2-be2f-567c11b9062a.PNG)

```
/*********
FIELD_FK_DOMAIN
all standard concept ids are part of specified domain
Parameters used in this template:
cdmDatabaseSchema = cdm1
vocabDatabaseSchema = cdm1
cdmTableName = EPISODE
cdmFieldName = EPISODE_OBJECT_CONCEPT_ID
fkDomain = Procedure OR Regimen
**********/
SELECT num_violated_rows, CASE WHEN denominator.num_rows = 0 THEN 0 ELSE 1.0*num_violated_rows/denominator.num_rows END  AS pct_violated_rows, 
  denominator.num_rows as num_denominator_rows
FROM
(
	SELECT COUNT(violated_rows.violating_field) AS num_violated_rows
	FROM
	(
		SELECT 'EPISODE.EPISODE_OBJECT_CONCEPT_ID' AS violating_field, cdmTable.* 
		  FROM cdm1.EPISODE cdmTable
		  LEFT JOIN cdm1.concept co
		    ON cdmTable.EPISODE_OBJECT_CONCEPT_ID = co.concept_id
		 WHERE co.concept_id != 0 AND co.domain_id NOT IN ('Procedure OR Regimen')
	) violated_rows
) violated_row_count,
( 
	SELECT COUNT(*) AS num_rows
	FROM cdm1.EPISODE cdmTable
) denominator
;
```

Is this the correct place where I should report these?

**Comment by Ozeriy** [comment]
Hi I would like to join the conversation regarding this error:
_**For the combination of CONCEPT_ID  3019800 (TROPONIN T.CARDIAC [MASS/VOLUME] IN SERUM OR PLASMA) and UNIT_CONCEPT_ID     8842 (NANOGRAM PER MILLILITER), the number and percent of records that have a value less than 1.00E+00. (Threshold=5%).**_
for the results of this test the abnormal range can be less then 1, and still I got fails. 
and I have that kind of DQD fails. it is possible it is also a bug?

**Comment by katy-sadowski** [comment]
Hi there, I've put up a PR to fix these issues.  As you'll see in the PR, I decided to remove that domain check altogether. The description of the episode_object_concept_id field in the threshold file states: 
```
Episode entries from the 'Disease Episode' concept class should have an episode_object_concept_id that comes from the Condition domain.  Episode entries from the 'Treatment Episode' concept class should have an episode_object_concept_id that come from the 'Procedure' domain or 'Regimen' concept class.
```
So to properly check this domain we'd additionally need to look at the episode concept class, something DQD is not equipped to do at the moment.

If you'd really like to include this check (as written) you could try a "hack" in the threshold file, writing the list of accepted domains as `Procedure', 'Regimen', 'Condition`.  But as-is this check could still allow inconsistent data to slip through, where the episode object's domain doesn't match the episode's class.

**Comment by katy-sadowski** [comment]
> Hi I would like to join the conversation regarding this error: _**For the combination of CONCEPT_ID 3019800 (TROPONIN T.CARDIAC [MASS/VOLUME] IN SERUM OR PLASMA) and UNIT_CONCEPT_ID 8842 (NANOGRAM PER MILLILITER), the number and percent of records that have a value less than 1.00E+00. (Threshold=5%).**_ for the results of this test the abnormal range can be less then 1, and still I got fails. and I have that kind of DQD fails. it is possible it is also a bug?

@Ozeriy - the latest threshold files have removed most of the measurement value checks (including this one). We are working on a larger re-evaluation of these thresholds and a process for setting the default values in a more robust & systematic way.  In the meantime, feel free to edit the plausible value in your local threshold file to whatever you believe should be plausible for your dataset.

**Comment by katy-sadowski** [comment]
This has been fixed in the v2.2.0 release. Please comment here if you're still experiencing issues with the latest DQD version!
